clients/wallet/src/resubmissions.rs

```diff
@@ -0,0 +1,795 @@
+
+		// Remove from cache
+		self.remove_tx_envelope_from_cache(tx.seq_num);
+
+		Err(ResubmissionError("Transaction already submitted".to_string()))
+	}
+
+	async fn bump_sequence_number_and_submit(
+		&self,
+		tx: Transaction,
+	) -> Result<TransactionResponse, Error> {
+		let sequence_number = self.get_sequence().await?;
+		let mut updated_tx = tx.clone();
+		updated_tx.seq_num = sequence_number + 1;
+
+		let old_tx_xdr = tx.to_base64_xdr();
+		let old_tx = String::from_utf8(old_tx_xdr.clone()).unwrap_or(format!("{old_tx_xdr:?}"));
+		tracing::trace!("bump_sequence_number_and_submit(): old transaction: {old_tx}");
+
+		let updated_tx_xdr = updated_tx.to_base64_xdr();
+		let updated_tx_xdr =
+			String::from_utf8(updated_tx_xdr.clone()).unwrap_or(format!("{updated_tx_xdr:?}"));
+		tracing::trace!("bump_sequence_number_and_submit(): new transaction: {updated_tx_xdr}");
+
+		let envelope = self.sign_and_create_envelope(updated_tx)?;
+		self.submit_transaction(envelope).await
+	}
+
+	/// This function iterates over all transactions of an account to see if a similar transaction
+	/// i.e. a transaction containing the same memo was already submitted previously.
+	/// TODO: This operation is very costly and we should try to optimize it in the future.
+	async fn is_transaction_already_submitted(&self, tx: &Transaction) -> bool {
+		// loop through until the 10th page
+		let mut remaining_page = 10;
+		let own_public_key = self.public_key();
+
+		while let Ok(transaction) = self.get_all_transactions_iter().await {
+			if remaining_page == 0 {
+				break
+			}
+
+			for response in transaction.records {
+				// Make sure that we are the sender and not the receiver because otherwise an
+				// attacker could send a transaction to us with the target memo and we'd wrongly
+				// assume that we already submitted this transaction.
+				let Ok(source_account) = response.source_account() else {
+                    continue
+                };
+				if !source_account.eq(&own_public_key) {
+					continue
+				}
+
+				// Check that the transaction contains the memo that we want to send.
+				let Some(response_memo) = response.memo_text() else { continue };
+				let Memo::MemoText(tx_memo) = &tx.memo else { continue };
+
+				if are_memos_eq(response_memo, tx_memo.get_vec()) {
+					return true
+				}
+			}
+			remaining_page -= 1;
+		}
+
+		// We did not find a transaction that matched our criteria
+		false
+	}
+}
+
+#[cfg_attr(test, mockable)]
+fn are_memos_eq(memo1: &Vec<u8>, memo2: &Vec<u8>) -> bool {
+	memo1 == memo2
+}
+
+#[cfg_attr(test, mockable)]
+async fn pause_process_in_secs(in_secs: u64) {
+	sleep(Duration::from_secs(in_secs)).await;
+}
+
+fn decode_to_envelope(
+	envelope_xdr_as_str_opt: &Option<String>,
+) -> Result<TransactionEnvelope, Error> {
+	let Some( envelope_xdr) = envelope_xdr_as_str_opt else {
+        tracing::warn!("handle_error(): no envelope_xdr found");
```

Comment:
Maybe we should also add the `envelope_xdr_as_str_opt` to the log here

Reply:
it's inside the `else{}` block, so `envelope_xdr_as_str_opt` is None, even if we log it